src/Sarif/Notes.cs

```diff
@@ -21,11 +21,15 @@ public static void LogNotApplicableToSpecifiedTarget(IAnalysisContext context, s
 
             // '{0}' was not evaluated for check '{1}' because the analysis
             // is not relevant for the following reason: {2}.
-            context.Logger.Log(context.Rule,
+
+            var rule = context.Rule.DeepClone();
+
+            context.Logger.Log(rule,
                 RuleUtilities.BuildResult(ResultKind.NotApplicable, context, null,
                     nameof(SdkResources.NotApplicable_InvalidMetadata),
+                    reportingDescriptor: rule,
```

Comment:
>reportingDescriptor: rule [](start = 20, length = 25)

I don't understand why you'd need to provide this object here after cloning. ?

Reply:
I did this because the rule and the context are changing. So, when I cloned the rule I had to use it... I'm not sure as well if that is the best way...

Reply:
Eddy, looks like you're not using the GH CodeFlow chrome extension, is that so?

---
In reply to: [550514500](https://github.com/microsoft/sarif-sdk/pull/2207#discussion_r550514500) [](ancestors = 550514500)

Reply:
reverted!

---
In reply to: [550511491](https://github.com/microsoft/sarif-sdk/pull/2207#discussion_r550511491) [](ancestors = 550511491)

Reply:
I was in the phone, so used the github interface. but yeah, i know the codeflow extension :)
thanks!

---
In reply to: [550532276](https://github.com/microsoft/sarif-sdk/pull/2207#discussion_r550532276) [](ancestors = 550532276,550514500)